### ISSUE TITLE
Localization support

### ISSUE THREAD

**Issue by anarchivist** [projectblacklight/arclight#544]
Support switching between languages in the UI.

**Comment by ggeisler** [comment]
Just to start figuring out details:

### How is a language made available?
In Blacklight, I believe any locale file in `config/locales` is used to populate the language-switcher menu available for end-users in the UI.

In Spotlight, the exhibit administrator explicitly selects one or more languages (from those for which a locale file exists, via an admin UI) to populate the language-switcher menu available for end-users in the UI.

ArcLight is more similar to Blacklight than Spotlight, and we don't currently have an Admin UI in ArcLight (though we do have mockups for one), so I assume it makes sense to go with the Blacklight approach?

### How do we display the language-switcher?
Blacklight shows the list of languages available using just the language codes (see the menu on the left, below), while in Spotlight (menu on the right) we enhanced that by showing each language using its native spelling/presentation:

 
![Screen Shot 2019-07-23 at 12 14 40 PM](https://user-images.githubusercontent.com/101482/61740476-8f4b4680-ad43-11e9-8b6e-88878a036144.png)

Can (technically) and should we use the enhanced, Spotlight approach to displaying the language-switcher menu, given it is more user-friendly?

**Comment by mejackreed** [comment]
Yes, @ggeisler I believe we can do the enhanced label approach. And perhaps could do that upstream in Blacklight if we wanted to.

**Comment by ggeisler** [comment]
👍 Yeah, unless there's some downside or significant overhead to doing this in Blacklight that I'm not aware of, it seems like we might as well do it there so the broader community also benefits.

**Comment by labradford** [comment]
Left notes on https://github.com/projectblacklight/arclight/pull/566
if someone else wants to work on this

**Comment by camillevilla** [comment]
@cbeer and I picked this up yesterday but are both out 8/16. 

We compared several approaches to implementing a locale picker:
- [Blacklight demo site](https://github.com/projectblacklight/demo.projectblacklight.org/blob/master/app/views/shared/_locale_picker.html.erb)
- [Spotlight](https://github.com/projectblacklight/spotlight/blob/master/app/views/spotlight/shared/_locale_picker.html.erb)
- Lea Ann's start at this in #566

Though the Blacklight demo site has localization implemented, an adopter still has to implement the locale picker themselves and choose how they want to approach routing. 

We created a gem ([blacklight-locale_picker](https://github.com/projectblacklight/blacklight-locale_picker)) for generating the locale picker and configuring which languages are available. 
Notes: 
- an implementer still needs to install the gem and choose what routing scheme best suits them (e.g. `/en/page` or `page?locale=en`). 
- locales are translated to language names, but we only had the list of English language names handy.

Unfortunately, we didn't have enough time to test this gem yesterday. It would be great if someone could try adding this to ArcLight's gemspec and testing out the generator in a new branch.
